### PR TITLE
docs(react-common): add a localization concept guide

### PR DESCRIPTION
**Pull request by ushkal**

### CODE REVIEW

File: packages/dx-react-common/docs/guides/localization.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+Complete code for this example is available at the following [link](https://devexpress.github.io/devextreme-reactive/react/grid/docs/guides/filtering/#custom-filter-operations).
+
+## Dynamic Data
+
+Some string contain text in a curly brackets. They are placeholders that are used to display some dynamic value. For example, the following string 'info' from the PagingPanel plugin "`${from}-${to} of ${count}`" contains 3 placeholders that will be replaced with actual data on rendering.
+
+
+See also:  
+[Grid localization]()  
+[Scheduler localization]()
```
Comment (MaximKudriavtsev) [inline packages/dx-react-common/docs/guides/localization.md]: Links

File: packages/dx-react-common/docs/guides/localization.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+/>
+<TableFilterRow
+  messages={{ month: 'Month equals' }}
+/>
+```
+Complete code for this example is available at the following [link](https://devexpress.github.io/devextreme-reactive/react/grid/docs/guides/filtering/#custom-filter-operations).
+
+## Dynamic Data
+
+Some string contain text in a curly brackets. They are placeholders that are used to display some dynamic value. For example, the following string 'info' from the PagingPanel plugin "`${from}-${to} of ${count}`" contains 3 placeholders that will be replaced with actual data on rendering.
```
Comment (RomanTsukanov) [inline packages/dx-react-common/docs/guides/localization.md]: IMPORTANT: There's only two backticks at the end of the code sample. This is the only way I could make GitHub recognize this text as a suggestion. Please add a third backtick.

```suggestion
Messages can contain texts in curly braces. These are placeholders used to display dynamic values. For example, the `PagingPanel` plugin's `info` message contains three placeholders:

```jsx
const pagingMessages = {
  info: '${from}-${to} of ${count}',
};
``
They will be replaced with real data when the plugin is rendered.
```

File: packages/dx-react-common/docs/guides/localization.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+
+See also:  
+[Grid localization]()
```
Comment (RomanTsukanov) [inline packages/dx-react-common/docs/guides/localization.md]:
```suggestion
[Grid Localization]()  
```

File: packages/dx-react-common/docs/guides/localization.md (hunk 1 above)
Comment (RomanTsukanov) [inline packages/dx-react-common/docs/guides/localization.md]:
```suggestion
[Scheduler Localization]()
```

File: packages/dx-react-common/docs/guides/localization.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
+# React Components - Localization
+
+This article describes how to localize UI texts in DevExtreme React Components.
+
+## The "Messages" Property
+
+Plugins that render UI texts have the `messages` property. It accepts an object that contains messages as key-value pairs in which keys are message IDs and values are the messages. For a list of message IDs that a plugin supports, see the Localization Messages section of the plugin's API Reference.
+
+To translate a message, find the plugin the renders it and specify the plugin's `messages` property with an object that contains your translations. Those message IDs that you omit from the object use default translations.
```
Comment (dirk-pieterse) [inline packages/dx-react-common/docs/guides/localization.md]: "plugin the renders" do you mean 'that' or 'which'?

File: packages/dx-react-common/docs/guides/localization.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
+# React Components - Localization
+
+This article describes how to localize UI texts in DevExtreme React Components.
+
+## The "Messages" Property
+
+Plugins that render UI texts have the `messages` property. It accepts an object that contains messages as key-value pairs in which keys are message IDs and values are the messages. For a list of message IDs that a plugin supports, see the Localization Messages section of the plugin's API Reference.
```
Comment (dirk-pieterse) [inline packages/dx-react-common/docs/guides/localization.md]:
```suggestion
Plugins that render UI texts have the `messages` property. It accepts an object that contains messages as key-value pairs in which keys are message IDs and values are messages. For a list of supported message IDs, see the Localization Messages section of the plugin's API Reference.
```

File: packages/dx-react-common/docs/guides/localization.md (hunk 4 above)
Comment (dirk-pieterse) [reply packages/dx-react-common/docs/guides/localization.md]:
```suggestion
To translate a message, find the plugin that renders it and specify the plugin's `messages` property with an object that contains your translations. Messages without IDs use default translations.
```

File: packages/dx-react-common/docs/guides/localization.md [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+To translate a message, find the plugin the renders it and specify the plugin's `messages` property with an object that contains your translations. Those message IDs that you omit from the object use default translations.
+
+The following code translates a message from the `Table` plugin.
```
Comment (dirk-pieterse) [inline packages/dx-react-common/docs/guides/localization.md]:
```suggestion
The following code translates a message from the `Table` plugin:
```

File: packages/dx-react-common/docs/guides/localization.md [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+/>
+<TableFilterRow
+  messages={{ month: 'Month equals' }}
+/>
+```
+The complete code is available in the [Custom Filter Operations](https://devexpress.github.io/devextreme-reactive/react/grid/docs/guides/filtering/#custom-filter-operations) article.
+
+## Placeholders
+
+Messages can contain texts in curly braces. These are placeholders used to display dynamic values. For example, the `PagingPanel` plugin's `info` message contains three placeholders:
```
Comment (dirk-pieterse) [inline packages/dx-react-common/docs/guides/localization.md]:
```suggestion
Messages can contain texts in curly braces. These placeholders are used to display dynamic values. For example, the `PagingPanel` plugin's `info` message contains three placeholders:
```

File: packages/dx-react-common/docs/guides/localization.md [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+
+```jsx
+const pagingMessages = {
+  info: '${from}-${to} of ${count}',
+};
+```
+They will be replaced with real data when the plugin is rendered.
```
Comment (dirk-pieterse) [inline packages/dx-react-common/docs/guides/localization.md]:
```suggestion
They are replaced with actual data when the plugin is rendered.
```

File: packages/dx-react-common/docs/guides/localization.md (hunk 7 above)
Comment (RomanTsukanov) [reply packages/dx-react-common/docs/guides/localization.md]: The original word order is correct.

File: packages/dx-react-common/docs/guides/localization.md (hunk 4 above)
Comment (RomanTsukanov) [reply packages/dx-react-common/docs/guides/localization.md]: > "plugin the renders" do you mean 'that' or 'which'?

You guessed correctly.

> Messages without IDs use default translations.

There cannot be a message without an ID. There can only be a message whose ID is not included in the object specified by the developer.

Comment (RomanTsukanov) [reply packages/dx-react-common/docs/guides/localization.md]:
```suggestion
To translate a message, find the plugin that renders it and specify the plugin's `messages` property with an object that contains your translations. Those message IDs that you omit from the object use default translations.
```

**Pull request merged**